main.js

```diff
@@ -52,10 +52,42 @@ function RoundToHundredths(x) {
   return Math.floor(x * 100 + 0.5) / 100;
 }
 
-function HandleNode(svgNode, scaleX, scaleY, translateX, translateY) {
+// |fillString| is expected to be "#RRGGBB" or "#RGB".
+function PathColorFromFill(fillString) {
+  if (fillString.length === 4) {
+    // Color in form of #RGB so let's turn that to #RRGGBB.
+    fillString = `#${fillString[1]}${fillString[1]}${fillString[2]}${fillString[2]}${fillString[3]}${fillString[3]}`.toUpperCase();
+  }
+
+  const r = fillString.substr(1,2);
+  const g = fillString.substr(3,2);
+  const b = fillString.substr(5,2);
+  
+  return `PATH_COLOR_ARGB, 0xFF, 0x${r}, 0x${g}, 0x${b},\n`;
+}
+
+function HandleNode(svgNode, scaleX, scaleY, translateX, translateY, preserveFill) {
   var output = '';
   for (var idx = 0; idx < svgNode.children.length; ++idx) {
     var svgElement = svgNode.children[idx];
+    const fill = svgElement.getAttribute('fill');
+
+    if (preserveFill && svgElement.tagName !== 'g' && fill && fill !== 'none') {
```

Comment:
Are there other tag names besides circle, path, g, and rect that we might run into? We'd probably rather avoid handling them at all, rather than just handling their color. You could pull this new stanza into a helper function that gets called from the circle, rect, & path branches. Or, move the <g> handling to the start of this function (out of the switch statement), and add a <default: return> to the switch so the output that's computed here is discarded.

Reply:
Good point, I've added the helper GetPathColorCommandFromFill which can be used on the current SVG elements for which a fill is relevant. Can be used on future elements added in the future.